internal/impl/ockam/input_kafka.go

```diff
@@ -0,0 +1,150 @@
+package ockam
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"github.com/benthosdev/benthos/v4/internal/impl/kafka"
+	"github.com/benthosdev/benthos/v4/public/service"
+)
+
+// this function is, almost, an exact copy of the init() function in ../kafka/input_kafka_franz.go
+func init() {
+	err := service.RegisterBatchInput("ockam_kafka", ockamKafkaInputConfig(),
+		func(conf *service.ParsedConfig, mgr *service.Resources) (service.BatchInput, error) {
+			i, err := newOckamKafkaInput(conf, mgr)
+			if err != nil {
+				return nil, err
+			}
+			return service.AutoRetryNacksBatchedToggled(conf, i)
+		})
+	if err != nil {
+		panic(err)
+	}
+}
+
+func ockamKafkaInputConfig() *service.ConfigSpec {
+	return kafka.FranzKafkaInputConfig().
+		Summary(`Ockam`).
+		Field(service.NewStringField("ockam_identity_name").Optional()).
+		Field(service.NewStringField("ockam_node_address").Default("127.0.0.1:6262")).
+		Field(service.NewStringField("ockam_allow_producer").Default("self")).
+		Field(service.NewStringField("ockam_enrollment_ticket").Optional()).
+		Field(service.NewStringField("ockam_relay").Optional())
+}
```

Comment:
I'll push some documentation for these tomorrow with some examples on how to use them.